addons/mail/static/src/core/common/thread.js

```diff
@@ -275,10 +274,20 @@ export class Thread extends Component {
         });
         onWillDestroy(() => stopOnChange());
         const saveScroll = () => {
-            this.props.thread.scrollTop =
-                ref.el.scrollHeight - ref.el.scrollTop - ref.el.clientHeight < 30
-                    ? "bottom"
-                    : ref.el.scrollTop;
+            let isBottom;
+            if (this.props.order === "asc") {
+                isBottom = ref.el.scrollHeight - ref.el.scrollTop - ref.el.clientHeight < 30;
+            } else {
+                isBottom = ref.el.scrollTop < 30;
```

Comment:
I feel like we are mixing up two concepts. Sometimes "bottom" means the bottom of the scroll-able container. Sometimes "bottom" refers to the "bottom" of the thread i.e. were the newest messages stand.